aocd/examples.py

```diff
@@ -91,13 +94,11 @@ def __getattr__(self, name):
             # actually used by an example parser
             raise AttributeError(name)
         article = self.article_a if part == "a" else self.article_b
-        if tag == "li":
-            # list items usually need further drill-down
-            result = article.find_all("li")
-            for li in result:
-                li.codes = [code.text for code in li.find_all("code")]
-        else:
-            result = [t.text for t in article.find_all(tag)]
+        assert article is not None
+        tags: list[bs4.Tag] = article.find_all(tag)
+        result: Union[list[bs4.Tag], list[str]] = tags
+        if tag != "li":
+            result = [t.text for t in tags]
```

Comment:
@cj81499  What happened here? It looks like the drill-down logic for li.codes got lost.

Reply:
Not entirely sure what I was thinking removing this. I'll put it back.

Any suggestions re: adding a unit test for this?

Reply:
Yes, was surprised that there was no test failure here!  Guess I'd [accidentally covered the lines](https://app.codecov.io/gh/wimglenn/advent-of-code-data/blob/main/aocd%2Fexamples.py) without actually testing the logic. It should be possible to tweak [`test_li_drilldown`](https://github.com/wimglenn/advent-of-code-data/blob/b968f40e19629ee7cc8a53423f0be71c90133ed6/tests/test_example_parser.py#L41-L47) into revealing a failure here.

Reply:
Hmm, actually, the existing test looks okay?!  It seems like it may have been nerfed in https://github.com/wimglenn/advent-of-code-data/commit/622e7838e0439d31188aec9cb145f091290222f1#diff-a6f2f29b66cb03c40b1693b351e42eaa9fe6f280bab27d6732ddf8a21d49db1b

Reply:
I think I originally removed it b/c I didn't see anything that uses `.codes` other than that test. Clearly, I was wrong and _something_ uses it (https://github.com/wimglenn/aocd-example-parser?).

I don't fully understand what's going on here, so I put the logic back to how it was before (with the exception of type-checking related changes).

Reply:
Yes, you found the right place - they're used by the locators in `aocd-example-parser`. Spooky action at a distance..